homeassistant/components/lamarzocco/config_flow.py

```diff
@@ -0,0 +1,242 @@
+
+    except AuthFail:
+        _LOGGER.error("Server rejected login credentials")
+        raise InvalidAuth
+    except RequestNotSuccessful:
+        _LOGGER.error("Failed to connect to server")
+        raise CannotConnect
+
+
+class LmConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for La Marzocco."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Init config flow."""
+        self._discovered: dict[str, str] = {}
+        self.reauth_entry: ConfigEntry | None
+        self._config: dict[str, Any] = {}
+        self._machines: list[tuple[str, str]] = []
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+
+        errors = {}
+
+        if user_input is not None:
+            data = user_input.copy()
+            data |= self._discovered
+            data[CONF_PORT] = DEFAULT_PORT_LOCAL
+            data[CONF_CLIENT_ID] = DEFAULT_CLIENT_ID
+            data[CONF_CLIENT_SECRET] = DEFAULT_CLIENT_SECRET
+
+            try:
+                self._machines = await get_machines(self.hass, data)
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except NoMachines:
+                errors["base"] = "no_machines"
+
+            if not errors:
+                self._config = data
+                return await self.async_step_machine_selection()
+
+        return self.async_show_form(
+            step_id="user", data_schema=LOGIN_DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_machine_selection(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Let user select machine to connect to."""
+        errors = {}
+        if user_input is not None:
+            serial_number = user_input[CONF_MACHINE]
+            await self.async_set_unique_id(serial_number)
+            self._abort_if_unique_id_configured()
+
+            # if host is set, check if we can connect to it
+            if user_input.get(CONF_HOST):
+                lm = LaMarzoccoClient(self.hass, self._config)
+                if not await lm.check_local_connection(
+                    credentials=self._config,
+                    host=user_input[CONF_HOST],
+                    serial=serial_number,
+                ):
+                    errors[CONF_HOST] = "cannot_connect"
+            if not errors:
+                return self.async_create_entry(
+                    title=serial_number, data=self._config | user_input
+                )
+
+        machine_options = [
+            SelectOptionDict(
+                value=serial_number,
+                label=f"{model_name} ({serial_number})",
+            )
+            for serial_number, model_name in self._machines
+        ]
+        machine_selection_schema = vol.Schema(
```

Comment:
Selecting a device from multiple makes no sense to me when Bluetooth initializes the config flow.

Reply:
The problem is, that you cannot rely on Bluetooth alone. So even if I discover the machine through Bluetooth I still need to supplement with the cloud data. A user normally only has one account for all its machines, so you need to select which machine that is.

Reply:
good point. Machine is now auto-selected when discovered